### DISCUSSION TITLE
Trigger Saving Files Save Dialog

### DISCUSSION THREAD

**Discussion by mytc** [quasarframework/quasar#9081] [General - Components / Directives / etc]
Hi,
I would like to saving a file into local storage in mobile app. This file is not from server, it is local generate file, thus it will not trigger the browser download behavior. 
How do I trigger the download event?
I have tried `exportFile` util but is not working in mobile. 
I want able to trigger the Save Dialog upon saving file event fired. 
q-file only allow to select files but can't saving it. 
Any idea?
Or workaround to get the selected folder/directory path?

**Comment by ontwikkelfabriek** [comment]
storing files in the local storage is not a very common way to `save` files...  Are you using Cordova or Capacitor for your mobile app?

**Reply by mytc** [reply]
Cordova. 
This is what I saw in an app that I would like to implement:
![savedialog](https://user-images.githubusercontent.com/3824518/116588292-f1d86a00-a94d-11eb-9478-5558092b7a1d.jpg)
I don't know what stake this app is using though.

**Reply by ontwikkelfabriek** [reply]
@mytc
If you want access to the filesystem to read/write files, you'll need this plugin:
https://cordova.apache.org/docs/en/10.x/reference/cordova-plugin-file/

**Reply by mytc** [reply]
Yes, I am using the file plugin right now. I can specific the directory to save the file but the problem still persist where user can't choose where user want to save.

**Reply by ontwikkelfabriek** [reply]
>  I can specific the directory to save 

How do you do that in your code? Apps on web browsers and on mobile devices can't save a file anywhere you( the app, or users) like.

**Reply by mytc** [reply]
by using the https://cordova.apache.org/docs/en/10.x/reference/cordova-plugin-file/ , I didn't save it anywhere I store it at 'file:///storage/emulated/0/Download'  directory.

**Reply by metalsadman** [reply]
you can probably leverage on the q-file's event handler, grab the file object there then use cordova's file plugin to save it wherever you want in your app. I use to do this with camera capture and save the file to a designated folder, but then i changed it to be saved within the app's sandbox. certainly there's an api in the cordova file plugin to save it to an external path of your choosing, then maybe have those external paths appear in a dialog where it is displayed as options then user chooses and it will return the path for the plugin to save.
